libs/application-generic/src/services/analytic-logs/workflow-run/workflow-run.schema.ts

```diff
@@ -54,6 +55,9 @@ const schemaDefinition = {
 
   // Data retention
   expires_at: { type: CHDateTime64(3, 'UTC') },
+
+  severity: { type: CHNullable(CHString()) }, // severity of the workflow run
+  critical: { type: CHNullable(CHBoolean()) }, // critical flag of the workflow run
```

Comment:
we can make those not nullable wit default values, it is more performant in ClickHouse to have any other value than null if possible.

```suggestion
  severity: { type: CHLowCardinality(CHString('none')) }, // severity of the workflow run
  critical: { type:CHBoolean(false) }, // critical flag of the workflow run
```

ALTER TABLE workflow_runs 
ADD COLUMN IF NOT EXISTS severity String DEFAULT 'none';

ALTER TABLE workflow_runs 
ADD COLUMN IF NOT EXISTS critical Boolean DEFAULT false';

Reply:
@djabarovgeorge, we can, but it will require us to migrate all the previously created workflow runs, as the type will indicate that there is always a value.

Reply:
The code wont handle "none" with backwards compatible way? Same for not critical.

What migration do you mean we will need?

Reply:
For the existing `workflow_runs` records, it will populate with the default value, for example, `none` but it might not be the correct value for that record; that's why we need the migration for the old records to fill it with the correct `severity, critical` values

Reply:
you’re totally right, i forgot we already deployed and started updating the workflow entities.
anyway, i’d love us to find a way to avoid creating nullable fields if possible. i can think of a few approaches without needing a migration:

1. set these fields with defaults. since they’re early-version fields that not many users rely on, it shouldn’t impact much, especially since they’re monitoring-related and not transactional.
2. for filtering queries like severity === VALUE && createdAt > DEPLOY_DATE, (we can add the date condition in the if statment). for the workflow run GET, we can map those default values to nullish.
3. merge the pr but defer the ui changes under a flag, and enable them in a couple of weeks once we already have some data stored in the database.